src/convertToObject.js

```diff
@@ -1,12 +1,23 @@
 'use strict';
 
-/**
+/*
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const styleUpgrated = sourceString
+    .split(';')
+    .map((el) => el.split(':'))
+    .reduce((prev, oneStyle) => {
+      if (oneStyle[0] && oneStyle[1]) {
+        return { ...prev, [oneStyle[0].trim()]: oneStyle[1].trim() };
+      }
+
+      return prev;
+    }, {});
```

Comment:
[CODE KNOWLEDGE] - There's no need to create a new object on each iteration of reduce. This can be inefficient for a large number of styles. Instead, modify the accumulator directly.